salt/utils/win_dacl.py

```diff
@@ -1115,34 +1121,37 @@ def get_name(principal):
         salt.utils.win_dacl.get_name('S-1-5-32-544')
         salt.utils.win_dacl.get_name('adminisTrators')
     '''
-    # If None is passed, use the Universal Well-known SID for "Null SID"
-    if principal is None:
-        principal = 'S-1-0-0'
-
-    # Assume PySID object
-    sid_obj = principal
-
-    # Name
-    try:
-        sid_obj = win32security.LookupAccountName(None, principal)[0]
-    except (pywintypes.error, TypeError):
-        pass
-
-    # String SID
-    try:
-        sid_obj = win32security.ConvertStringSidToSid(principal)
-    except (pywintypes.error, TypeError):
-        pass
+    # If this is a PySID object, use it
+    if isinstance(principal, pywintypes.SIDType):
+        sid_obj = principal
+    else:
+        # If None is passed, use the Universal Well-known SID for "Null SID"
+        if principal is None:
+            principal = 'S-1-0-0'
+
+        # String SID
+        # Try Converting String SID to SID Object first as it's least expensive
+        try:
+            sid_obj = win32security.ConvertStringSidToSid(principal)
+        except pywintypes.error:
+
+            # Name
+            # Try Getting the SID Object by Name Lookup last
+            # This is expensive, especially on large AD Domains
+            try:
```

Comment:
Right after the `except` statement, the blank line and `# Name` probably need to be removed (or `# Name` elaborated on).